base/range.jl

```diff
@@ -115,9 +124,47 @@ convert(::Type{T}, r::AbstractRange) where {T<:AbstractRange} = r isa T ? r : T(
 
 ## ordinal ranges
 
+"""
+    OrdinalRange{T, S} <: AbstractRange{T}
+
+Supertype for ordinal ranges with elements of type `T` with
+spacing(s) of type `S`.
+An ordinal range is one whose values have a well-defined ordering.
+The spacing of the elements of the range does not need to be even.
+[`UnitRange`](@ref), [`StepRange`](@ref), and other types are subtypes of this.
+"""
 abstract type OrdinalRange{T,S} <: AbstractRange{T} end
+
+"""
+    AbstractUnitRange{T} <: OrdinalRange{T, T}
+
+Supertype for unit ranges with elements of type `T`.
```

Comment:
I think I'd prefer ``ranges with a step size of [`oneunit`](@ref)` `` over "unit ranges".